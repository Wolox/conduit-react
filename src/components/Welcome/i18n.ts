import i18next from 'i18next';

i18next.addResources('es', 'Welcome', {
  description: 'Un lugar para compartir su conocimiento en React'
});

i18next.addResources('en', 'Welcome', {
  description: 'A place to share your React knowledge'
});
