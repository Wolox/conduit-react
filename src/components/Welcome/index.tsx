import { memo } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './styles.module.scss';

function Welcome() {
  const { t } = useTranslation('Welcome');
  return (
    <div className={styles.contentWelcome}>
      <h1 className={styles.title}>{t('conduit')}</h1>
      <p className={styles.description}>{t('description')}</p>
    </div>
  );
}

export default memo(Welcome);
