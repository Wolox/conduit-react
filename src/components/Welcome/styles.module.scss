@import 'scss/variables/_colors';
@import 'scss/fonts/_fonts';

.content-welcome {
  background-color: $green-fern;
  box-shadow: 0 8px 8px -8px $gray-shadow;
  margin-bottom: 32px;
  padding: 32px;
  text-align: center;

  .title {
    color: $white;
    display: block;
    font-family: $titillium;
    font-size: 56px;
    font-weight: 600;
    line-height: initial;
    padding-bottom: 8px;
    text-shadow: 0 1px 3px $gray-shadow;
  }

  .description {
    color: $white;
    font-size: 24px;
    font-weight: 400;
    line-height: initial;
  }
}
