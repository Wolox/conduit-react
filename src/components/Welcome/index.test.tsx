import { render } from '@testing-library/react';

import Welcome from '.';

describe('Component Welcome', () => {
  test('renders component welcome', () => {
    const { asFragment } = render(<Welcome />);
    expect(asFragment()).toMatchSnapshot();
  });
});
