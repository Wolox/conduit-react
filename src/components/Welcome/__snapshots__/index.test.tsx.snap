// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Welcome renders component welcome 1`] = `
<DocumentFragment>
  <div
    class="contentWelcome"
  >
    <h1
      class="title"
    >
      conduit
    </h1>
    <p
      class="description"
    >
      description
    </p>
  </div>
</DocumentFragment>
`;
